### PR TITLE
Change push-ci-build.sh to push-build.sh.

### PR DESCRIPTION
**Pull request by david-mcmahon**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32235)

### CODE REVIEW

**Comment by ixdy** [comment]
How do we coordinate this with https://github.com/kubernetes/release/pull/66?

**Comment by david-mcmahon** [comment]
As `build.sh` conveniently clones it right before using it, it's safe to get the companion change in right before this one goes in.  Admittedly this is a bit hard to do with the SQ, but maybe @k8s-oncall could help.

**Comment by david-mcmahon** [comment]
@k8s-bot retest issue #UNKNOWN

**Comment by david-mcmahon** [comment]
@rmmh @ixdy @spxtr 
I'm not sure what's going on with this PR failing.  Is it some fallout from something you are doing or might it have something to do with the state of the repo on the test machine and that fact that this PR contains a renamed file?  I'm grasping here b/c I'm not sure what's going on.

File: hack/jenkins/build.sh [hunk 1]
```diff
@@ -67,26 +61,25 @@ make release
...
+  
+  push_build=${release_infra_clone}/push-build.sh
 
-  if [[ ! -x ${release_infra_clone}/push-ci-build.sh ]]; then
-    echo "FATAL: Something went wrong." \
-         "${release_infra_clone}/push-ci-build.sh isn't available." \
-         "Exiting..." >&2
-    exit 1
+  if [[ ! -x ${push_build} ]]; then
+    # TODO: Remove/Restore this with the full deprecation PR
```
Comment (ixdy) [inline hack/jenkins/build.sh]: with this new logic, this shouldn't be an if/else, just an if.

File: hack/jenkins/build.sh [hunk 2]
```diff
@@ -40,14 +40,8 @@ export CLOUDSDK_COMPONENT_MANAGER_DISABLE_UPDATE_CHECK=true
...
-# New kubernetes/release/push-ci-build.sh values
-# RELEASE_INFRA_PUSH=true when we're using kubernetes/release/push-ci-build.sh
-: ${RELEASE_INFRA_PUSH:="false"}
-# SET_NOMOCK_FLAG=true means we're doing full pushes and we pass --nomock to 
-# push-ci-build.sh. This is set to false in the
-# testing jobs and only used in the RELEASE_INFRA_PUSH=true scope below.
-: ${SET_NOMOCK_FLAG:="true"}
-export KUBE_RELEASE_RUN_TESTS RELEASE_INFRA_PUSH FEDERATION SET_NOMOCK_FLAG
+# New kubernetes/release/push-build.sh values
+export KUBE_RELEASE_RUN_TESTS FEDERATION
```
Comment (ixdy) [inline hack/jenkins/build.sh]: do we need to export FEDERATION? or is it only used for the `--federation` flag below?

**Comment by ixdy** [comment]
GitHub still thinks there are merge conflicts, which is probably why this is failing.

**Comment by ixdy** [comment]
LGTM

@k8s-bot test this, issue #27462

**Pull request merged**